client/packages/programs/src/JsonForms/components/Search/useSearchQueries.tsx

```diff
@@ -1,69 +1,55 @@
-import React, { useState } from 'react';
-import {
-  Typography,
-  useTranslation,
-  RegexUtils,
-} from '@openmsupply-client/common';
-import { usePatient } from '@openmsupply-client/system';
+import { useState } from 'react';
+import { RegexUtils, FilterBy } from '@openmsupply-client/common';
+import { PatientRowFragment, usePatient } from '@openmsupply-client/system';
 
-export const QueryValues = ['patientByCode'] as const;
+export const QueryValues = ['patientSearch'] as const;
 type QueryValue = (typeof QueryValues)[number];
 
-type GetDisplayElement = (
-  result: Record<string, unknown>
-) => JSX.Element | null;
-
 interface SearchQueryOptions {
   query?: QueryValue;
   optionString?: string;
   displayString?: string;
   saveFields?: string[];
-  placeholderText?: string;
 }
 
 interface SearchQueryOutput {
-  runQuery: (searchValue: string) => void;
+  runQuery: (searchFilter: FilterBy) => void;
   saveFields: string[] | null;
-  getOptionLabel?: (result: Record<string, unknown>) => string;
-  getDisplayElement?: GetDisplayElement;
-  placeholderText: string;
+  getOptionLabel: (result: PatientRowFragment) => string;
 }
 
-const { formatTemplateString, removeEmptyLines } = RegexUtils;
+const { formatTemplateString } = RegexUtils;
 
 export const useSearchQueries = ({
   query,
```

Comment:
This `query` parameter could be removed because there is only one query type, right? and then you can also simplify the `searchQueries` below?

Reply:
Currently there is only one type ("patientSearch"). However, the original intention was that this component could be used for any type of search, so we can add more "query" types later.

Reply:
This is removed now as per [this update](https://github.com/openmsupply/open-msupply/pull/2216#discussion_r1333711998)